### PR TITLE
Prevent invalidating table check constraints when altering table schema

### PR DESCRIPTION
**Pull request by fulghum**
Updated logic for dropping and renaming columns to ensure that no table checks are referenced. 

Further improvements should eventually be made to ensure that table check constraints aren't modified by other schema alterations, such as modifying a column's type.

### CODE REVIEW

File: enginetest/enginetests.go [hunk 1]
```diff
@@ -1699,6 +1698,21 @@ func TestRenameColumn(t *testing.T, harness Harness) {
 		[]sql.Row{
 			{1, "first row"},
 		}, nil, nil)
+
+	// Test column rename behavior when a table check constraint is involved
+	RunQuery(t, e, harness, "ALTER TABLE mytable ADD CONSTRAINT test_check CHECK (i2 < 12345)")
```
Comment (zachmu) [inline enginetest/enginetests.go:1703]: Should enclose this new section in a named `t.Run()`

File: enginetest/enginetests.go [hunk 2]
```diff
@@ -1699,6 +1698,21 @@ func TestRenameColumn(t *testing.T, harness Harness) {
...
+	AssertErr(t, e, harness, "ALTER TABLE mytable RENAME COLUMN i2 TO i3", sql.ErrCheckConstraintInvalidatedByColumnAlter)
+	RunQuery(t, e, harness, "ALTER TABLE mytable RENAME COLUMN s2 TO s3")
+	tbl, ok, err = db.GetTableInsensitive(NewContext(harness), "mytable")
```
Comment (zachmu) [inline enginetest/enginetests.go:1706]: Use empty lines to break up these assertions into logical groupings

File: enginetest/enginetests.go [hunk 3]
```diff
@@ -1970,6 +1984,22 @@ func TestDropColumn(t *testing.T, harness Harness) {
 
 		TestQueryWithContext(t, ctx, e, "ALTER TABLE mydb.tabletest DROP COLUMN s", []sql.Row(nil), nil, nil)
 	})
+
+	// Test drop column behavior when a table check constraint is involved
+	RunQuery(t, e, harness, "ALTER TABLE mytable ADD COLUMN j int, ADD COLUMN k int")
```
Comment (zachmu) [inline enginetest/enginetests.go:1989]: Same comment about `t.Run`

File: sql/analyzer/check_constraints.go [hunk 4]
```diff
@@ -142,8 +142,9 @@ func loadChecks(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (sql.No
 
 			return &nn, nil
 		case *plan.Update:
-			rtable := getResolvedTable(node)
+			nn := *node
```
Comment (zachmu) [inline sql/analyzer/check_constraints.go:145]: Fine to move this up but replace all the `node` references below with `nn`.

Same with other blocks

File: sql/analyzer/check_constraints.go [hunk 5]
```diff
@@ -29,6 +27,8 @@ import (
 
 // validateCreateCheck legal expressions for CREATE CHECK statements, including those embedded in CREATE TABLE
 // statements
+// TODO: validateCreateCheck doesn't currently do any type validation on the check and will allow you to create
+//       checks that will never evaluate correctly.
 func validateCreateCheck(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope) (sql.Node, error) {
```
Comment (zachmu) [inline sql/analyzer/check_constraints.go:32]: Probably should be named validateCheckConstraints, and get an updated comment about it applying to many DDL statements

File: sql/analyzer/validate_create_table.go [hunk 6]
```diff
@@ -90,11 +94,18 @@ func validateAlterColumn(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope
...
+			sch, err = validateDropColumn(initialSch, sch, n)
+			if err != nil {
+				return nil, err
+			}
 		}
 		return n, nil
 	})
 }
 
+// Schema is passed in twice, because one version is the initial version before the alter column expressions are
```
Comment (zachmu) [inline sql/analyzer/validate_create_table.go:107]: Start golang function comments with the name of the function and a one-sentence description of what it does

File: sql/analyzer/validate_create_table.go [hunk 7]
```diff
@@ -146,6 +173,36 @@ func validateModifyColumn(intialSch sql.Schema, schema sql.Schema, mc *plan.Modi
...
+
+	return newSch, nil
+}
+
+func validateDropColumn(initialSch, sch sql.Schema, dc *plan.DropColumn) (sql.Schema, error) {
+	table := dc.Child
+	nameable := table.(sql.Nameable)
+
+	// Iterate over checks and make sure none of them reference dc.Column
+	for _, check := range dc.Checks {
```
Comment (zachmu) [inline sql/analyzer/validate_create_table.go:188]: Extract a method for this and the other identical block?

**Review by zachmu — APPROVED** [review]
LGTM, just a few comments mostly about structure

File: enginetest/enginetests.go (hunk 1 above)
Comment (fulghum) [reply enginetest/enginetests.go:1703]: ✅

File: enginetest/enginetests.go (hunk 2 above)
Comment (fulghum) [reply enginetest/enginetests.go:1706]: ✅

File: enginetest/enginetests.go (hunk 3 above)
Comment (fulghum) [reply enginetest/enginetests.go:1989]: ✅

File: sql/analyzer/check_constraints.go (hunk 5 above)
Comment (fulghum) [reply sql/analyzer/check_constraints.go:32]: ✅

File: sql/analyzer/validate_create_table.go (hunk 6 above)
Comment (fulghum) [reply sql/analyzer/validate_create_table.go:107]: ✅

File: sql/analyzer/validate_create_table.go (hunk 7 above)
Comment (fulghum) [reply sql/analyzer/validate_create_table.go:188]: ✅

File: sql/analyzer/check_constraints.go (hunk 4 above)
Comment (fulghum) [reply sql/analyzer/check_constraints.go:145]: ✅

**Pull request merged**